pkg/client.go

```diff
@@ -259,6 +260,62 @@ func (c *BlobCacheClient) GetContent(hash string, offset int64, length int64) ([
 	return nil, ErrContentNotFound
 }
 
+func (c *BlobCacheClient) GetContentStream(hash string, offset int64, length int64) (chan []byte, error) {
+	ctx, cancel := context.WithTimeout(c.ctx, getContentRequestTimeout)
+	contentChan := make(chan []byte)
+
+	go func() {
+		defer close(contentChan)
+		defer cancel()
+
+		for attempt := 0; attempt < 3; attempt++ {
+			client, host, err := c.getGRPCClient(ctx, &ClientRequest{
+				rt:   ClientRequestTypeRetrieval,
+				hash: hash,
+			})
+			if err != nil {
+				return
+			}
+
+			stream, err := client.GetContentStream(ctx, &proto.GetContentRequest{Hash: hash, Offset: offset, Length: length})
+			if err != nil {
+				c.metadata.RemoveEntryLocation(ctx, hash, host)
+				c.mu.Lock()
+				delete(c.localHostCache, hash)
+				c.mu.Unlock()
+				continue
+			}
+
+			for {
+				resp, err := stream.Recv()
+				if err != nil {
+					if err == io.EOF {
+						return
+					}
+
+					c.metadata.RemoveEntryLocation(ctx, hash, host)
+					c.mu.Lock()
+					delete(c.localHostCache, hash)
+					c.mu.Unlock()
+					break
+				}
+
+				if !resp.Ok {
+					c.metadata.RemoveEntryLocation(ctx, hash, host)
+					c.mu.Lock()
+					delete(c.localHostCache, hash)
+					c.mu.Unlock()
+					break
+				}
+
+				contentChan <- resp.Content
+			}
```

Comment:
This could be cleaned up a bit to be 
```go
    resp, err := stream.Recv()
    if err == io.EOF {
        return
    }
    
    if err != nil || !resp.Ok {
        c.metadata.RemoveEntryLocation(ctx, hash, host)
        c.mu.Lock()
        delete(c.localHostCache, hash)
        c.mu.Unlock()
        break
    }

    contentChan <- resp.Content
```

Also, should we log the error here/can you help me understand the cleanup we're doing? Might be worth adding a comment for clarity down the line.